src/components/editor/sidebar.tsx

```diff
@@ -0,0 +1,56 @@
+import { Cog8ToothIcon } from "@heroicons/react/24/solid";
+import { EllipsisVerticalIcon } from "@heroicons/react/24/solid";
+
+function PageSelect({
+  page,
+  selectedPage,
+  setSelectedPage,
+}: {
+  page: any;
+  selectedPage: string;
+  setSelectedPage: any;
+}) {
+  const isSelected = selectedPage === page.id;
+
+  return (
+    <div
+      className={`p-2 m-2 cursor-pointer rounded-md ${isSelected && "highlighted"}`}
+      onClick={() => {
+        setSelectedPage(page.id);
+      }}
+    >
+      <div className="flex justify-between items-center">
+        <p className="text-lg">{page.title}</p>
+        <a href="#" className="text-right"><EllipsisVerticalIcon className="h-5 w-5" /></a>
+      </div>
+    </div>
+  );
+}
+
+export function Sidebar({
```

Comment:
I wonder if we could manage state differently using objects here. Seems like the perfect case for that. 

I'm really just thinking out loud here, no need to do any changes.